### ISSUE TITLE
All contribution and development documentation should live in GitHub

### ISSUE THREAD

**Issue by nrmancuso** [checkstyle/checkstyle#14542]
Right now, we have development and contribution documentation both on our website and GitHub. I think it makes sense to consolidate these to GitHub and allow our website to be only for Checkstyle usage. To be clear, i am proposing that we move the entire "Developers" section to GitHub.

This has several advantages:
1. Code blocks are rendered more clearly in markdown
2. Markdown supports different types of syntax highlighting
3. Code blocks are easily copy-pasted

I really can't think of any disadvantages for this. Personally, I find markdown much easier to read and much prefer markdown for developer documentation due to it's simple formatting.

If we move forward with this issue, we should also remove the IDE setup stuff from the wiki since it will now live in GitHub (we should probably just do this anyway now).

**Comment by rnveach** [comment]
I think it makes sense to move the following:
* https://checkstyle.org/contributing.html
* https://checkstyle.org/beginning_development.html

There is another unlisted pages that should also consider to be split/move. It has no left nav, which should be an issue as everything should be in the left nav or else you can't navigate to it.
* https://checkstyle.org/report_issue.html

I think https://checkstyle.org/extending.html should stay as I consider it how to use checkstyle as a library and matching to version jar and web site publication. The other sections are how to use our pre-existing checks, but we are also a library and some of our users use as to create custom checks/ASTs since we are so hard to allow new ones.

**Comment by nrmancuso** [comment]
> I think it makes sense to move the following:
> 
> * https://checkstyle.org/contributing.html
> * https://checkstyle.org/beginning_development.html
> 
> There is another unlisted pages that should also consider to be split/move. It has no left nav, which should be an issue as everything should be in the left nav or else you can't navigate to it.
> 
> * https://checkstyle.org/report_issue.html
> 
> I think https://checkstyle.org/extending.html should stay as I consider it how to use checkstyle as a library. The other sections are how to use our pre-existing checks, but we are also a library and some of our users use as to create custom checks/ASTs since we are so hard to allow new ones.

I fully agree with all points here.

**Comment by nrmancuso** [comment]
@romani if you are good, please place approved label on here

**Comment by romani** [comment]
yes , Github is so famous now and all know how to work with it and ...... , that docs in github is more beneficial.

**Comment by SURYANSHUAGRAWAL2006** [comment]
Hello sir, I fullly undertood this issue just wanted to confirm that, should we migrate the content from src/site/xdoc/contributing.xml, beginning_development.xml, and related non-release-versioned pages into Markdown files in the repository root (e.g., CONTRIBUTING.md), remove the xdoc pages, and update site navigation accordingly?
Or would you prefer keeping a minimal page on the website that redirects to GitHub?

**Comment by romani** [comment]
Considering how much this pages already spread at world, redirection is required.

**Comment by SURYANSHUAGRAWAL2006** [comment]
Hello Sir, I noticed that `contributing.xml` and `CONTRIBUTING.md` contain different content and structure.
Before proceeding with redirection, should we merge all useful content from the website page into the Markdown version so that no information is lost?
Or is it acceptable to keep the GitHub version as the single source of truth even if some sections are removed?

**Comment by SURYANSHUAGRAWAL2006** [comment]
Hello @romani please tell me so that I can continue to solve this issue.
thank u sir.

**Comment by romani** [comment]
Please create PR and in PR will discuss what is missing or Missing on purpose

**Comment by SURYANSHUAGRAWAL2006** [comment]
Hello sir, please review it if it has any problem then I'll update.
thank u sir.

**Comment by SURYANSHUAGRAWAL2006** [comment]
Hello Sir, please review my changes so, that I can continue forward.
Thank u sir.

**Comment by romani** [comment]
contributing is migrated https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/2254c64_20260224161630/contributing.html#

Still to be done:
https://checkstyle.org/beginning_development.html
https://checkstyle.org/report_issue.html

**Comment by SURYANSHUAGRAWAL2006** [comment]
Hello sir, I also want to show development section into github but I have a question as contributing.md file is present to show contributing section into that, but any file related to development not present so, `should new file to be make related to development to show beginning_development.xml section into that file `please answered me so, that I can start work as soon as possible.
Thank u sir.

**Comment by SURYANSHUAGRAWAL2006** [comment]
please sir reply me I would only like to confirm that my approach is correct or not so, that I can start work early.
Thank u sir.

**Comment by romani** [comment]
I don't understand your post, please rephrase it.

In short:
https://checkstyle.org/report_issue.html
We should move all to report-issue.md , report_issue.html should reference to report-issue.md

**Comment by SURYANSHUAGRAWAL2006** [comment]
Hello Sir, I check it out there is no file named is report-issue.md so, should I make new file which will name `report-issue.md`.
Thank u sir.

**Comment by romani** [comment]
https://checkstyle.org/report_issue.html is migrated , @SURYANSHUAGRAWAL2006 did it.

**Comment by romani** [comment]
@SURYANSHUAGRAWAL2006 , please help to do final update for https://checkstyle.org/beginning_development.html

**Comment by SURYANSHUAGRAWAL2006** [comment]
Hello sir please respond me for link check issue as I replied u several times please help me.
Thak u sir.

**Comment by romani** [comment]
Please share link to question here, sorry that I can not manage all messages

**Comment by SURYANSHUAGRAWAL2006** [comment]
https://github.com/checkstyle/checkstyle/issues/11572

**Comment by romani** [comment]
Final fix is merged.
Our hero is @SURYANSHUAGRAWAL2006 .

**Comment by pbludov** [comment]
@romani Note that the website still has a link to the old location of the "issue report" page:

 https://checkstyle.org/report_issue.html points to https://github.com/checkstyle/checkstyle/blob/master/.github/report-issue.md instead of https://github.com/checkstyle/checkstyle/blob/master/docs/report-issue.md

**Comment by romani** [comment]
@pbludov , https://github.com/checkstyle/checkstyle/issues/20743 is created
